app/lib/annotation_viz_service.rb

```diff
@@ -111,17 +111,8 @@ def self.user_annot_to_annot(user_annotation, cluster)
   # returns a flat array of annotation objects, with name, scope, annotation_type, and values for each
   def self.available_annotations(study, cluster: nil, current_user: nil, annotation_type: nil)
     annotations = []
-    viewable = study.viewable_metadata
-    metadata = annotation_type.nil? ? viewable : viewable.select {|m| m.annotation_type == annotation_type}
-    metadata = metadata.map do |annot|
-      {
-        name: annot.name,
-        type: annot.annotation_type,
-        values: sanitize_values_array(annot.values, annot.annotation_type),
-        scope: 'study'
-      }
-    end
-    annotations.concat(metadata)
+    metadata_annots = available_metadata_annotations(study, annotation_type: annotation_type)
```

Comment:
I recently learned that in [Ruby 3.1](https://www.ruby-lang.org/en/news/2021/12/25/ruby-3-1-0-released/), values in hash literals and keyword arguments can be omitted.  So we could do `available_metadata_annotations(study, annotation_type:)`, pretty much like JavaScript.

With [Ruby 2.6 EOL in two months](https://endoflife.date/ruby), do you or @bistline happen to know of any dependencies that'd hold us back from upgrading to 3.1 as opposed to 2.7?

Reply:
Our Docker image uses [Phusion Passenger](https://github.com/phusion/passenger-docker) as the underlying layer.  They have yet to update to Ruby 3.1 (although they are at 3.0.3).  We'd need to wait for them to catch up, and then we'd also need to figure out the issues with updating to Ubuntu focal (20.04) and our base image build automation.  I tried to update the managed Google base image to 20.04, and I could not get it to build due to some really low-level weirdness with platform architecture.